### PR TITLE
Fixes Rolling Chairs over glass

### PR DESCRIPTION
**Pull request by ArcLumin**
This should fix all sorts of rolling objects that go over glass causing their user to take damage when they don't wear shoes, as seen in https://github.com/HippieStationCode/HippieStation13/issues/2749

### CODE REVIEW

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 1]
```diff
@@ -329,7 +329,7 @@
 		playsound(loc, 'sound/effects/glass_step.ogg', 50, 1)
 		if(ishuman(AM))
 			var/mob/living/carbon/human/H = AM
-			if(!(PIERCEIMMUNE in H.dna.species.specflags))
+			if(!(PIERCEIMMUNE in H.dna.species.specflags) && !buckled_mob,/obj/structure/bed/chair/withwheels)
```
Comment (Carbonhell) [inline code/game/objects/items/stacks/sheets/glass.dm]: change && to ||
and put !istype(buckled_mob,/obj/structure/bed/chair/withwheels)

Comment (Carbonhell) [inline code/game/objects/items/stacks/sheets/glass.dm]: also changelog

**Comment by ArcLumin** [comment]
coding on my phone is cancer

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 2]
```diff
@@ -329,7 +329,7 @@
 		playsound(loc, 'sound/effects/glass_step.ogg', 50, 1)
 		if(ishuman(AM))
 			var/mob/living/carbon/human/H = AM
-			if(!(PIERCEIMMUNE in H.dna.species.specflags))
+			if(!(PIERCEIMMUNE in H.dna.species.specflags) || istype(buckled_mob,/obj/structure/bed/chair/withwheels))
```
Comment (Carbonhell) [inline code/game/objects/items/stacks/sheets/glass.dm]: !istype not istype

Comment (ArcLumin) [inline code/game/objects/items/stacks/sheets/glass.dm]: FUCK ME

**Comment by Lawligagger** [comment]
ye

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 3]
```diff
@@ -329,7 +329,7 @@
 		playsound(loc, 'sound/effects/glass_step.ogg', 50, 1)
 		if(ishuman(AM))
 			var/mob/living/carbon/human/H = AM
-			if(!(PIERCEIMMUNE in H.dna.species.specflags))
+			if(!(PIERCEIMMUNE in H.dna.species.specflags) || !istype(buckled_mob,/obj/structure/bed/chair/withwheels))
```
Comment (JohnGinnane) [inline code/game/objects/items/stacks/sheets/glass.dm]: The issue I have with this is it only affects chairs with wheels and might not encompass all wheeled furniture you can sit/lie on. For example there the roller bed:

https://github.com/HippieStationCode/HippieStation13/blob/5462010e5bcc8a61ad5618ffd0f04f07eeb570b0/code/game/objects/structures/stool_bed_chair_nest/bed.dm#L84

If we have to add special cases each time then it gets kinda snowflakey. What if we checked the item H is buckled to has wheels using a new var on the wheeled furniture?

**Comment by ghost** [comment]
Yeah okay +1.

**Comment by ArcLumin** [comment]
Let's see if I can use buckled.anchored to cover more objects

**Comment by ArcLumin** [comment]
> undef var
> rip

**Comment by ArcLumin** [comment]
actually fuckin worked wew

**Comment by ArcLumin** [comment]
Now it should work for all buckled unanchored mobs

**Comment by ArcLumin** [comment]
This only needs a code review now. So, now to summon code reviewers

File: code/game/objects/items/stacks/sheets/glass.dm [hunk 4]
```diff
@@ -329,7 +329,7 @@
 		playsound(loc, 'sound/effects/glass_step.ogg', 50, 1)
 		if(ishuman(AM))
 			var/mob/living/carbon/human/H = AM
-			if(!(PIERCEIMMUNE in H.dna.species.specflags))
+			if(!(PIERCEIMMUNE in H.dna.species.specflags) || isobj(usr.buckled) && !usr.buckled.anchored)
```
Comment (JohnGinnane) [inline code/game/objects/items/stacks/sheets/glass.dm]: I think you'll wanted to wrap the `isobj(usr.buckled) && !usr.buckled.anchored` in ( and ) so it goes like this:
- Is the human immune to pierce?

**OR**
- is the user buckled to something **AND** is the buckled item not anchored

**Comment by SplergeBucket** [comment]
RESPOND TO THE CODE REVIEW AND HAVE IT ALL SANCTIONED AND I'LL SAY YES THOUGH I DO ADMIT I'D LIKE A BIT OF A PERCENTAGE CHANCE AS IF YOU'RE BEING DRAGGED WHILE BUCKLED YOU SHOULD HAVE A CHANCE, MAYBE LIKE 30% TO DODGE SOME GLASS SHARDS.

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 4 above)
Comment (ArcLumin) [reply code/game/objects/items/stacks/sheets/glass.dm]: Isn't that what it's doing already

**Comment by blobbernaut** [comment]
I LIKE YELLING TOO

**Comment by ArcLumin** [comment]
Why does this still have code review, and hasn't been merged?

File: code/game/objects/items/stacks/sheets/glass.dm (hunk 4 above)
Comment (TDcoolguy300) [reply code/game/objects/items/stacks/sheets/glass.dm]: He means another wrap which would end up looking like this
`if(!(PIERCEIMMUNE in H.dna.species.specflags) || (isobj(usr.buckled) && !usr.buckled.anchored))`

**Pull request merged**